dl/src/test/scala/com/intel/analytics/bigdl/nn/ModuleSpec.scala

```diff
@@ -126,4 +127,40 @@ class ModuleSpec extends FlatSpec with Matchers {
     module2.eq(module) should be(false)
   }
 
+  "getName for different module" should "return different name" in {
+    val module1 = new Linear[Double](2, 3)
+    val module2 = new Linear[Double](4, 5)
+
+    module1.getName() should not be (module2.getName())
```

Comment:
add another checking: if the two models is equal than they share the same hashcode as we've just changed the hashcode method?